code/aepcomposeui/src/main/java/com/adobe/marketing/mobile/aepcomposeui/aepui/AepUi.kt

```diff
@@ -0,0 +1,42 @@
+package com.adobe.marketing.mobile.aepcomposeui.aepui
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import com.adobe.marketing.mobile.aepuitemplates.AepUiTemplate
+import com.adobe.marketing.mobile.aepuitemplates.SmallImageTemplate
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.AepUiState
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.SmallImageUIState
+
+
+/**
+ * Represents a UI component that can be rendered in the AepUI Engine. Binds a template data with a tracking observer.
+ * This is a sealed interface that can be implemented by different UI components like [SmallImageAepUi], [LargeImageAepUi], etc.
+ * This allows restricting the type of template and observer that can be associated with a UI component, which can be later
+ * used in the app to render the UI component via UIComposeExtensions.
+ * @param T The type of the template associated with the UI component.
+ * @param S The type of the state associated with the UI component.
+ */
+sealed interface AepUI<T : AepUiTemplate, S : AepUiState> {
+    fun getTemplate(): T
+    fun getState(): S
+    fun updateState(newState: S)
+}
+
+
+class SmallImageAepUi(
```

Comment:
Can we move this to its own file? This file will get cluttered as we add more AepUIs.

Also can we decide if it needs to be `Aep` or `AEP` and apply it across all file and class names? I vote for the latter since its an acronym and it would be consistent with the class names in Core UI service